src/codechandler/CodecHandler.cpp

```diff
@@ -0,0 +1,48 @@
+#include "CodecHandler.h"
+
+bool CodecHandler::GetInformation(kodi::addon::InputstreamInfo& info)
+{
+  AP4_GenericAudioSampleDescription* asd(nullptr);
+  if (m_sampleDescription)
+  {
+    if ((asd = dynamic_cast<AP4_GenericAudioSampleDescription*>(m_sampleDescription)))
+    {
+      if ((!info.GetChannels() && asd->GetChannelCount() != info.GetChannels()) ||
+          (!info.GetSampleRate() && asd->GetSampleRate() != info.GetSampleRate()) ||
+          (!info.GetBitsPerSample() && asd->GetSampleSize() != info.GetBitsPerSample()))
+      {
+        if (!info.GetChannels())
+          info.SetChannels(asd->GetChannelCount());
+        if (!info.GetSampleRate())
+          info.SetSampleRate(asd->GetSampleRate());
+        if (!info.GetBitsPerSample())
+          info.SetBitsPerSample(asd->GetSampleSize());
+        return true;
+      }
+    }
+    else
+    {
+      //Netflix Framerate
+      AP4_Atom* atom;
+      AP4_UnknownUuidAtom* nxfr;
+      static const AP4_UI08 uuid[16] = {0x4e, 0x65, 0x74, 0x66, 0x6c, 0x69, 0x78, 0x46,
```

Comment:
you are know to what is referred this fixed UUID? or from what come from?
i think that this is a good candidate to be moved as constexpr in a unnamed namespace on top of this file

Reply:
It's literally `NetflixFrameRate`
Maybe there's a nicer way via helper function to make this, if there's other examples that need it too

Reply:
how about:
```
AP4_Atom* atom;
      AP4_UnknownUuidAtom* nxfr;
      static const std::string uuid = "NetflixFrameRate";
      atom = m_sampleDescription->GetDetails().GetChild(
          reinterpret_cast<const AP4_UI08*>(uuid.c_str()));
      if (atom && (nxfr = dynamic_cast<AP4_UnknownUuidAtom*>(atom)) &&
          nxfr->GetData().GetDataSize() == 10)
```
Can then move uuid to unnamed namespace, easier to read

Reply:
it was so simple I hadn't thought...
leave it as it is for now we avoid conversions but move the uuid to unnamed namespace
suggest only to add in the constexpr variable a little comment to explain that the value is "NetflixFrameRate" for posterity
if we will find others similar cases we will think to create an utils method

Reply:
I've done my own test to see that the data is the same in both scenarios but if you feel it's safer I can revert and just move the variable as suggested?

Reply:
not so good, in an unnamed namespace, `static` it is irrilevant, and you must use `constexpr` on variables,
since `"NetflixFrameRate"` is already a `const char*` there is no reason to convert it into a string object to be re-converted as `const char*`, then should be defined as:

`constexpr const char* netflixFramerateUuid = "NetflixFrameRate";`

and remove `.c_str()` on reinterpreter cast, suggest to use also full name ^^